addons/rmsmartshape/shapes/shape_base.gd

```diff
@@ -787,33 +762,33 @@ func generate_collision_points() -> PackedVector2Array:
 	return points
 
 
-func bake_collision():
+func bake_collision() -> void:
 	if not has_node(collision_polygon_node_path):
 		return
-	var polygon = get_node(collision_polygon_node_path)
-	var points = generate_collision_points()
-	var transformed_points = PackedVector2Array()
-	var poly_transform = polygon.get_global_transform()
-	var shape_transform = get_global_transform()
+	var polygon: CollisionPolygon2D = get_node(collision_polygon_node_path)
```

Comment:
Use `get_node() as CollisionPolygon2D` just to be sure it's actually the correct node.